app/controllers/spree/admin/products_controller.rb

```diff
@@ -11,6 +11,12 @@ class ProductsController < ResourceController
       update.before :update_before
       helper_method :clone_object_url
 
+      def new
+        @product.available_on ||= Time.current
```

Comment:
@aleksandarpetrushev let's move that into a before action and not overwrite the new method - it should be cleaner :)